### PR TITLE
Upgrade ror and shakapacker

### PR DESCRIPTION
**Pull request by ahangarha**
Upgraded:
- Ruby to 3.1.2
- ReactOnRails to 13.3.2
- Shakapacker to 6.6.0

- - -
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails_demo_ssr_hmr/53)

### CODE REVIEW

File: bin/dev [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+rescue Errno::ENOENT
+  false
+end
+
+def run(process)
+  system "#{process} start -f Procfile.dev"
+rescue Errno::ENOENT
+  warn <<~MSG
+    ERROR:
+    Please ensure `Procfile.dev` exist in your project!
```
Comment (justin808) [inline bin/dev:15]: grammar

`exists`

File: bin/dev-static [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+rescue Errno::ENOENT
+  false
+end
+
+def run(process)
+  system "#{process} start -f Procfile.dev-static"
+rescue Errno::ENOENT
+  warn <<~MSG
+    ERROR:
+    Please ensure `Procfile.dev-static` exist in your project!
```
Comment (justin808) [inline bin/dev-static:15]: `exists`

File: config/webpack/serverWebpackConfig.js [hunk 3]
```diff
@@ -103,14 +105,14 @@ const configureServer = () => {
...
   // If using the React on Rails Pro node server renderer, uncomment the next line
   // serverWebpackConfig.target = 'node'
 
-  return serverWebpackConfig
-}
+  return serverWebpackConfig;
+};
 
-module.exports = configureServer
+module.exports = configureServer;
```
Comment (justin808) [inline config/webpack/serverWebpackConfig.js:118]: Why all the formatting changes? Are we adding settings for eslint and rubocop?

**Review by justin808 — CHANGES_REQUESTED** [review]

File: bin/dev (hunk 1 above)
Comment (ahangarha) [reply bin/dev:15]: My bad!
This should be fixed in ROR as well. (made pr there)

Should I fix it manually here or wait for another release on ROR and then update them here?

File: config/webpack/serverWebpackConfig.js (hunk 3 above)
Comment (ahangarha) [reply config/webpack/serverWebpackConfig.js]: Reverted

**Comment by ahangarha** [comment]
@justin808 

I updated and tested this PR again.
- ✅ SSR works
- ✅ HMR works

Also updated to Shakapacker 6.6.0

**Pull request merged**